src/Nethermind/Nethermind.TxPool/ITxPool.cs

```diff
@@ -35,7 +35,7 @@ public interface ITxPool
         void AddPeer(ITxPoolPeer peer);
         void RemovePeer(PublicKey nodeId);
         AddTxResult AddTransaction(Transaction tx, TxHandlingOptions handlingOptions);
-        void RemoveTransaction(Keccak hash, long blockNumber);
+        void RemoveTransaction(Keccak hash, long blockNumber, bool removeSmallerNonces);
```

Comment:
void RemoveTransactions(Address sander, long removeBelowThisNonce) and separate these two calls

Reply:
I really don't like this. It will make implementation harder, we don't have other use cases for it beyond what is here, it will require 2 calls from client. Instead I will rename to removeBelowThisTxNonce and make it false by default for explicit usage.